src/main/java/com/hartwig/platinum/config/version/PipelineVersion.java

```diff
@@ -6,7 +6,8 @@
 import java.util.List;
 
 public class PipelineVersion {
-    String padVersionStringToSemanticVersion(final String candidate) {
+    String padVersionStringToSemanticVersion(final String inputCandidate) {
+        String candidate = inputCandidate.replaceAll("-(alpha|beta).[0-9]+$", "");
```

Comment:
This feels a bit like adding a hack on top of a hack. Better to make the parsing a bit more solid:
`^(\d+)\.(\d+)\.(\d+)(?:-((?:alpha|beta)(?:\.[0-9]+)*))?$`

This regex is used everywhere else. Capturing group 1/2/3 are major/minor/patch. Capturing group 4 is optional and will match alpha or beta if it exists.

Your version will work as well, but this version will not match a bunch of false positives.

Reply:
Good point, made the change (without using group 4 as I don't think it's relevant for platinum logic).